### PR TITLE
Added arachnid

### PR DESCRIPTION
**Pull request by watzon**
Added arachnid. A powerful web scraping framework.

It doesn't have specs yet because of how difficult specs are to write for a project like this, and because I will be transitioning from a standard http client to a headless chrome based client as soon as I finish [watzon/marionette](https://github.com/watzon/marionette), so if you don't want to merge yet I understand.

### CODE REVIEW

**Comment by watzon** [comment]
Oh crap, I didn't realize the table of contents got changed. It was autogenerated and I'll fix it. As far as webdriver goes I'm trying to stay away from that because it relies on selenium which is really memory hungry and uses Java.

File: README.md [hunk 1]
```diff
@@ -315,6 +315,7 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [myhtml](https://github.com/kostya/myhtml) - Fast HTML5 Parser that includes CSS selectors
 
 ## HTTP
+ * [arachnid](https://github.com/watzon/arachnid) - Powerful web scraping framework for Crystal
```
[the remark this answers is not in the document]
Comment (watzon) [reply README.md]: Should be good now. Fixed the TOC as well.

**Comment by watzon** [comment]
True, I'm much better with integration tests than unit tests generally, but I'll try and get some added and then ping you here.

**Comment by veelenga** [comment]
@watzon i am close to merge this. Please add tests, periodical CI build and resolve conflicts.

**Pull request merged**